src/qtt/algorithms/markov_chain.py

```diff
@@ -179,7 +182,8 @@ def stationary_distribution_discrete(jump_chain) -> np.ndarray:
         return pi
 
     def generate_sequence(self, length: int, delta_time: float,
-                          initial_state: Union[None, int, np.ndarray] = None) -> np.ndarray:
+                          initial_state: Union[None, int, np.ndarray] = None,
+                          generators: Optional[List[ChoiceGenerator]] = None) -> np.ndarray:
```

Comment:
This parameter generators is added but to feed this generate sequence with an alternative generator than the default is difficult, it is tightly coupled to this class.
It is not wrong but I think it will not be used. Do you see a use case for it?

Reply:
I agree that it won't be used. It was added to eliminate the overhead of constructing generators on the fly. I considered making the generators a private attribute of the structure, but then we would have to check whether those generators match with the `delta_time` parameter.
If a future PR we could clean this up a bit and make the attributes like `jump_chain` read only.